conan/tools/intel/inteloneapi.py

```diff
@@ -0,0 +1,178 @@
+            return "Intel C++ Compiler 19.2"
+        elif self._mode == "icx":
+            return "Intel C++ Compiler %s" % (self._compiler_version.split('.')[0])
+        else:  # DPC++ compiler
+            return "Intel(R) oneAPI DPC++ Compiler"
+
+    def generate(self, group="build"):
+        if platform.system() == "Windows" and not self._conanfile.win_bash:
+            content = textwrap.dedent("""\
+                @echo off
+                {}
+                """.format(self.command))
+            filename = self.filename + '.bat'
+        else:
+            filename = self.filename + '.sh'
+            content = self.command
+        create_env_script(self._conanfile, content, filename, group)
+
+    @property
+    def installation_path(self):
+        installation_path = self._conanfile.conf["tools.intel:installation_path"] or \
+                            get_inteloneapi_installation_path()
+        self._out.info("Got Intel oneAPI installation folder: %s" % installation_path)
+        return installation_path
+
+    @property
+    def command(self):
+        """
+        The Intel oneAPI DPC++/C++ Compiler includes environment configuration scripts to
+        configure your build and development environment variables:
+            On Linux*, the file is a shell script called setvars.sh.
+            On Windows*, the file is a batch file called setvars.bat.
+
+        * Linux:
+            >> . /<install-dir>/setvars.sh <arg1> <arg2> … <argn><arg1> <arg2> … <argn>
+
+            The compiler environment script file accepts an optional target architecture
+            argument <arg>:
+                intel64: Generate code and use libraries for Intel 64 architecture-based targets.
+                ia32: Generate code and use libraries for IA-32 architecture-based targets.
+        * Windows:
+            >> call <install-dir>\\setvars.bat [<arg1>] [<arg2>]
+
+            Where <arg1> is optional and can be one of the following:
+                intel64: Generate code and use libraries for Intel 64 architecture
+                         (host and target).
+                ia32: Generate code and use libraries for IA-32 architecture (host and target).
+            With the dpcpp compiler, <arg1> is intel64 by default.
+
+            The <arg2> is optional. If specified, it is one of the following:
+                vs2019: Microsoft Visual Studio* 2019
+                vs2017: Microsoft Visual Studio 2017
+
+        :return: `str` setvars.sh|bat command to be run
+        """
+        # Let's check if user wants to use some custom arguments to run the setvars script
+        command_args = self._conanfile.conf["tools.intel:setvars_args"] or ""
+        # The setvars script is going to be loaded/cleared up every conanfile.run() execution
+        # but we will check this env variable just in case
+        if str(os.getenv("SETVARS_COMPLETED", "")) == "1" and "force" not in command_args:
```

Comment:
Maybe this is not the right place, because this command is not executed right now, but can be executed later.
I would probably add this check inside the generated script file, so it is checked right before doing the actual call to activate the intel environment (e.g. someone calling the activate script twice)

Reply:
Indeed, there is no need to have it there either. The `setvars` script has already a mechanism to handle it. Removing that check